### PR TITLE
Add python support in integration tests

### PR DESCRIPTION
**Pull request by maxday**
- rewrite the recorder extension in Go so it can be executed in both node and python runtimes (more coming ?)
- add python support
- add DD_SERVICE tests

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -61,4 +60,5 @@ jobs:
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           # Todo : retrieve this layer version automatically from AWS
           NODE_LAYER_VERSION: 55
+          PYTHON_LAYER_VERSION: 41
```
Comment (nhinsch) [inline .github/workflows/integration_tests.yml:63]: Do we still have a plan to automatically retrieve this from AWS?

File: integration_tests/recorder-extension/main.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	"sort"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/agent-payload/gogen"
+)
+
+var (
+	extensionName   = "recorder-extension" // extension name has to match the filename
```
Comment (nhinsch) [inline integration_tests/recorder-extension/main.go:29]: Nit: Any reason this is a `var` and not a `const`?

File: integration_tests/recorder-extension/main.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	}
+	res := NextEventResponse{}
+	err = json.Unmarshal(body, &res)
+	if err != nil {
+		return nil, err
+	}
+	return &res, nil
+}
+
+// Start begins running the sidecar
```
Comment (nhinsch) [inline integration_tests/recorder-extension/main.go:184]: What is "the sidecar" here?

File: integration_tests/recorder-extension/main.go [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+		for _, log := range messages {
+			sortedTags := strings.Split(log.Tags, ",")
+			sort.Strings(sortedTags)
+			log.Tags = strings.Join(sortedTags, ",")
+			jsonLog, err := json.Marshal(log)
+			if err != nil {
+				fmt.Printf("Error while JSON encoding the Log")
+			}
+			stringJsonLog := string(jsonLog)
+			// if we log an unwanted log, it will be availble in the next log api payload -> infinite loop
```
Comment (nhinsch) [inline integration_tests/recorder-extension/main.go:233]: Nit: availble -> available

File: integration_tests/recorder-extension/main.go [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+		}
+	})
+
+	http.HandleFunc("/api/v1/series", func(w http.ResponseWriter, r *http.Request) {
+	})
+
+	http.HandleFunc("/api/v1/check_run", func(w http.ResponseWriter, r *http.Request) {
+	})
+
+	// port 8080 is used by the Lambda Invoke API
```
Comment (nhinsch) [inline integration_tests/recorder-extension/main.go:246]: I would move this comment to where you specify the port as 3333.

File: integration_tests/run.sh [hunk 6]
```diff
@@ -38,26 +40,28 @@ trap remove_stack EXIT
...
 NODE_LAYER_VERSION=${NODE_LAYER_VERSION} \
+PYTHON_LAYER_VERSION=${PYTHON_LAYER_VERSION} \
 serverless deploy --stage ${stage}
 
 # invoking functions
-metric_function_names=("enhancedMetricTest" "noEnhancedMetricTest" "timeoutMetricTest")
-log_function_names=("logTest")
+metric_function_names=("enhanced-metric-node" "enhanced-metric-python" "no-enhanced-metric-node" "no-enhanced-metric-python" "timeout-node" "timeout-python")
+log_function_names=("log-node" "log-python")
+trace_function_names=()
```
Comment (nhinsch) [inline integration_tests/run.sh:49]: Why is this empty? What does it do?

**Review by nhinsch — APPROVED** [review]
Looks great! Just left a few comments.

File: .github/workflows/integration_tests.yml (hunk 1 above)
Comment (maxday) [reply .github/workflows/integration_tests.yml:63]: ✅

File: integration_tests/recorder-extension/main.go (hunk 2 above)
Comment (maxday) [reply integration_tests/recorder-extension/main.go]: not at all !

File: integration_tests/run.sh (hunk 6 above)
Comment (maxday) [reply integration_tests/run.sh:62]: yes empty for now, supported in #11

**Pull request merged**